### PR TITLE
Cursed Dice

### PR DESCRIPTION
**Pull request by Arthurdentist**
This is far from complete but the basic structure is there, feedback / criticism is welcome

Playing games with the devil is dangerous business and now you can do it too! Just find a Mysterious d20 during your time as a xenoarcheologist and see how well you do!

If you roll poorly then nasty things will happen to you. Score well and you will be rewarded!

To do:-
- [x] Test Thoroughly because this is probably already a buggy piece of shit with awful code (tested and hopefully not awful)
- [x] Rectify awful code
- [x] Make the dice xenoarcheology loot
- [x] Changelog
- [x] Tweak curses / rewards and perhaps add more variety. 

inb4 100 comments telling me how awful this whole thing is

### CODE REVIEW

**Comment by AutismSpider** [comment]
Curses range form turning into clown/mime (forces the suit on you etc) just harmless stuff?

**Comment by Arthurdentist** [comment]
roll of 1 = become a cluwne like the admin punishment mob cluwne
roll of 20 = rare minerals and a lot of good genetic powers

Anything from 2 - 9 = bad things
Roll of 10 - 12 = nothing happens
Roll of 13 = unlucky number, dice deactivates
Roll of 14 - 19 = good shit

I'm open for more suggestions for any of those categories. 

Also forgot the mention that the dice eventually deactivates itself even if a 13 isn't rolled. It can do this on the first roll but in my experience it tends to get a good three / four rolls in before it just becomes a normal dice. This stops you from spamming it ad infinitum

**Comment by AutismSpider** [comment]
how about for 10 to 12 they get random food or drink?

if they roll a 7 the dice should get some power back? or they get bag of gold coins?

**Comment by Intigracy** [comment]
Make 13 pop (boom) the dice and hurt the person who rolled it.

**Comment by Arthurdentist** [comment]
@AutismSpider  once its a normal dice its a normal dice. An admin would be able to re-activate it through var edit but outside of that I want deactivation to be fairly permanent

Giving small things on 10 to 12 isn't a bad idea though

@Intigracy: I like this idea, I will change it to this sometime tomorrow

**Comment by ihadtoregisterforthis** [comment]
I'd say keep the spells and descriptions fantasy boardgame related, since it's a d20 and all. A different effect for every number could be fun.

Will it have a different sprite?

**Comment by Zth--** [comment]
What if every person can use the dice only once? Or give the dice X number of uses (always once per person)

**Comment by Arthurdentist** [comment]
@Zth-- I'd like to give some absolute madman the opportunity to wreck themselves by rolling it a few times. I'm also quite attached to the randomness element of the deactivation element of the dice.

@ihadtoregisterforthis I wouldn't mind more curse/reward ideas but I don't want to tie specific curses / rewards to specific numbers (With the exceptions of 1, 13 and 20). I feel like that'd make it too predictable.

I've also never played DnD so I don't really know that much about DnD references.

File: code/game/objects/items/weapons/dice.dm [hunk 1]
```diff
@@ -122,3 +122,169 @@
...
+						user.dna.SetSEState(NOBREATHBLOCK,1)
+						genemutcheck(user,COLDBLOCK,null,MUTCHK_FORCED)
+						genemutcheck(user,FIREBLOCK,null,MUTCHK_FORCED)
+						genemutcheck(user,NOBREATHBLOCK,null,MUTCHK_FORCED)
+						user.update_mutations()
+						to_chat(user, "<span class=danger><B>You have been granted protection! </span></B>")
+					if(4)
+						getFromPool(/obj/item/stack/sheet/mineral/gold,user.loc,50)
+						to_chat(user, "<span class=danger)(B>You have been reward in silver! </span></B>")
+					if(5)
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/dice.dm]: These are reversed

**Comment by ghost** [comment]
If I read this right, if you're a tajaran who rolls 6-9 and pick lands on 3 for the species swap, you're missing an else punishment for being tajaran? Essentially makes it a get out of curse free if I understand this

File: code/game/objects/items/weapons/dice.dm [hunk 2]
```diff
@@ -122,3 +122,169 @@
...
+		return 0
+
+/obj/item/weapon/dice/d20/cursed/diceroll(mob/user as mob, thrown)
+	..()
+	if(deactivated == 0) //If the dice has power then something will happen
+		message_admins("[key_name(user)] has [thrown? "used" : "thrown"] a cursed dice and rolled a [result]")
+		log_game("[key_name(user)] has [thrown? "used" : "thrown"] a cursed dice and rolled a [result]")
+		if(istype(user,/mob/living/carbon/human))
+			var/mob/living/carbon/human/h = user
+			if(result == 1)
```
Comment (PJB3005) [inline code/game/objects/items/weapons/dice.dm]: Make this gigantic tree of if statements into switch statements.

With a switch, you can do `if(x to y)`.

File: code/game/objects/items/weapons/dice.dm [hunk 3]
```diff
@@ -122,3 +122,169 @@
 				for(var/obj/machinery/computer/bhangmeter/bhangmeter in doppler_arrays)
 					if(bhangmeter)
 						bhangmeter.sense_explosion(epicenter.x,epicenter.y,epicenter.z,round(uncapped*0.25), round(uncapped*0.5), round(uncapped),"???", cap)
+
+
+
+/obj/item/weapon/dice/d20/cursed
+	desc = "Something about this dice seems wrong"
+	name = "Mysterious d20"
```
Comment (PJB3005) [inline code/game/objects/items/weapons/dice.dm]: Needs `\improper`

**Comment by FudgePucker** [comment]
Seven should give you a cookie instead of bad shit.

Seven isn't an unlucky number

**Comment by Arthurdentist** [comment]
okay, IF has become SWITCH, gold and silver reward text is fixed and the name is now /improper

more testing still required.

File: code/game/objects/items/weapons/dice.dm [hunk 4]
```diff
@@ -122,3 +122,170 @@
...
+							genemutcheck(user,BLINDBLOCK,null,MUTCHK_FORCED)
+							user.update_mutations()
+							to_chat(user, "<span class=danger><B>You've gone blind, deaf and mute! </span></B>")
+						if(2)
+							for(var/datum/organ/external/l_arm/E in h.organs)
+								E.droplimb(1)
+							for(var/datum/organ/external/r_arm/E in h.organs)
+								E.droplimb(2)
+						if(3)
+							if(h.species.name != "Tarjaran") //someone who was made a catbeast by the dice can't become a different species by getting lucky
```
Comment (KingDairy) [inline code/game/objects/items/weapons/dice.dm]: "Tarjaran"
there's an extra "r" in there

**Comment by Arthurdentist** [comment]
Fixed catbeast mispelling, made the dice explode on a roll of 13 and added it to xenoarcheology.

**Comment by Arthurdentist** [comment]
changelog

**Comment by ghost** [comment]
COMPRESS COMMITS

**Comment by Arthurdentist** [comment]
fuck sake I've bollocked it up

**Pull request closed without merging**